akka-multi-node-testkit/src/main/scala/akka/remote/testkit/MultiNodeSpec.scala

```diff
@@ -160,29 +160,43 @@ object MultiNodeSpec {
   require(selfName != "", "multinode.host must not be empty")
 
   /**
-   * Port number of this node. Defaults to 0 which means a random port.
+   * TCP Port number to be used when running tests on TCP. 0 means a random port.
    *
    * {{{
-   * -Dmultinode.port=0
+   * -Dmultinode.tcp.port=0
    * }}}
    */
-  val selfPort: Int = Integer.getInteger("multinode.port", 0)
+  val tcpPort: Int = Integer.getInteger("multinode.tpc.port", 0)
 
-  require(selfPort >= 0 && selfPort < 65535, "multinode.port is out of bounds: " + selfPort)
+  require(tcpPort >= 0 && tcpPort < 65535, "multinode.tcp.port is out of bounds: " + tcpPort)
 
   /**
-   * UDP Port number to be used on this node. 0 means a random port.
+   * UDP Port number to be used when running tests on UDP. 0 means a random port.
    *
    * {{{
-   * -Dmultinode.udp-port=0
+   * -Dmultinode.udp.port=0
    * }}}
    */
-  val udpPort: Option[Int] = Option(System.getProperty("multinode.udp-port")) match {
-    case None    => None
-    case Some(_) => Some(Integer.getInteger("multinode.udp-port", 0))
-  }
+  val udpPort: Option[Int] =
+    Option(System.getProperty("multinode.udp.port")).map { _ =>
+      Integer.getInteger("multinode.udp.port", 0)
+    }
+
+  require(udpPort.getOrElse(1) >= 0 && udpPort.getOrElse(1) < 65535, "multinode.udp.port is out of bounds: " + udpPort)
+
+  /**
+   * Port number of this node.
+   *
+   * This is defined in function of property `multinode.protocol`.
+   * If set to 'udp', udpPort will be used. If unset or any other value, it will default to tcpPort.
+   */
+  val selfPort: Int =
+    System.getProperty("multinode.protocol") match {
+      case "udp" => udpPort.getOrElse(0)
+      case _     => tcpPort
+    }
 
-  require(udpPort.getOrElse(1) >= 0 && udpPort.getOrElse(1) < 65535, "multinode.udp-port is out of bounds: " + udpPort)
+  require(selfPort >= 0 && selfPort < 65535, "multinode.port is out of bounds: " + selfPort)
```

Comment:
it's not `multinode.port` now

Reply:
I removed this check now. `selfPort` is either the `tcpPort` or the `udpPort` and those two have already their own `require()` call.